modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/SpeechToTextServiceImpl.java

```diff
@@ -214,11 +220,23 @@ protected OrganizationDirectoryService getOrganizationDirectoryService() {
     return organizationDirectoryService;
   }
 
-  @Reference
+  //TODO: Improve OSGI wiring to change STT Engine implementation during OC execution
+  @Reference(
+          policy = ReferencePolicy.DYNAMIC,
+          policyOption = ReferencePolicyOption.GREEDY,
+          cardinality = ReferenceCardinality.MULTIPLE,
+          target = "(engineType=vosk)",
+          unbind = "unsetSpeechToTextEngine",
+          bind = "setSpeechToTextEngine"
+  )
```

Comment:
It would be best if you enforced the Reference name with `name = "SpeechToTextEngine"`. You do not need cardinality `MULTIPLE`, `MANDATORY` should be sufficient. Use the policy `STATIC` instead of `DYNAMIC`. This will ensure that the component gets deactivated if the bound service is unavailable.

Reply:
Now works! No more `nullpointer error` when the parameter changes during execution